From 3e3c4fa34b5e51e2436ae81faaf1fad90c136d39 Mon Sep 17 00:00:00 2001
From: KyrinCode <kyrincode@qq.com>
Date: Sat, 13 Sep 2025 00:50:53 +0800
Subject: [PATCH 1/2] skip reorg check after regenesis

---
 synchronizer/synchronizer.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/synchronizer/synchronizer.go b/synchronizer/synchronizer.go
index 5708756..96088b0 100644
--- a/synchronizer/synchronizer.go
+++ b/synchronizer/synchronizer.go
@@ -602,6 +602,13 @@ hash and has parent. This operation has to be done until a match is found.
 */
 func (s *ClientSynchronizer) checkReorg(latestStoredBlock etherman.Block, syncedBlock *etherman.Block) (*etherman.Block, error) {
 	// This function only needs to worry about reorgs if some of the reorganized blocks contained rollup info.
+
+	// Skip reorg check after a regenesis
+	if latestStoredBlock.BlockNumber < s.genBlockNumber {
+		log.Debugf("NetworkID: %d, skipping reorg check after a regenesis. Latest stored block: %d, genesis block: %d", s.networkID, latestStoredBlock.BlockNumber, s.genBlockNumber)
+		return nil, nil
+	}
+
 	latestStoredEthBlock := latestStoredBlock
 	reorgedBlock := latestStoredBlock
 	var depth uint64
-- 
2.39.5 (Apple Git-154)

